File: CodeForces/Codeforces_938/D_Inaccurate_Subsequence_Search.cpp
// Link : https://codeforces.com/contest/1955/problem/D

#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, m, k;
        cin >> n >> m >> k;
        
        vector<int> a(n);
        vector<int> b(m);
        
        for (int i = 0; i < n; ++i) {
            cin >> a[i];
        }
        
        for (int i = 0; i < m; ++i) {
            cin >> b[i];
        }
        
        unordered_map<int, int> dicta;
        unordered_map<int, int> dictb;
        
        for (int i : b) {
            dictb[i]++;
        }
        
        for (int i = 0; i < m; ++i) {
            dicta[a[i]]++;
        }
        
        int ans = 0;
        int c = 0;
        
        for (const auto& kv : dictb) {
            c += min(dicta[kv.first], kv.second);
        }
        
        if (c >= k) {
            ans++;
        }
        
        for (int i = m; i < n; ++i) {
            dicta[a[i - m]]--;
            if (dictb[a[i - m]] > dicta[a[i - m]]) {
                c--;
            }
            dicta[a[i]]++;
            if (dictb[a[i]] >= dicta[a[i]]) {
                c++;
            }
            if (c >= k) {
                ans++;
            }
        }
        
        cout << ans << endl;
    }
    
    return 0;
}

File: CodeForces/Codeforces_938/D_Inaccurate_Subsequence_Search.py
# Link : https://codeforces.com/contest/1955/problem/D

import sys
from collections import defaultdict, deque
from heapq import heappop, heappush
import math
import bisect
import itertools
import functools
import random

def main():
    t = int(input())
    for _ in range(t):
        n, m, k = map(int, input().split())
        a = list(map(int, input().split()))
        b = list(map(int, input().split()))
        
        dicta = defaultdict(int)
        dictb = defaultdict(int)
        
        for i in b:
            dictb[i] += 1
            
        for i in range(m):
            dicta[a[i]] += 1
            
        ans = 0
        c = 0
        
        for key in dictb:
            c += min(dicta[key], dictb[key])
            
        if c >= k:
            ans += 1
            
        for i in range(m, n):
            dicta[a[i - m]] -= 1
            if dictb[a[i - m]] > dicta[a[i - m]]:
                c -= 1
            dicta[a[i]] += 1
            if dictb[a[i]] >= dicta[a[i]]:
                c += 1
            if c >= k:
                ans += 1
                
        print(ans)


if __name__ == "__main__":
    main()